client/keys/add.go

```diff
@@ -204,7 +206,19 @@ func runAddCmd(_ *cobra.Command, args []string) error {
 
 	if len(mnemonic) == 0 {
 		// read entropy seed straight from crypto.Rand and convert to mnemonic
-		entropySeed, err := bip39.NewEntropy(mnemonicEntropySize)
+
+		// get the mnemonic size
+		mnemonicSize := viper.GetInt(flagMnemonicSize)
+		if mnemonicSize == 0 {
+			mnemonicSize = 24
```

Comment:
I would rather remove this. If users passed `--mnemonic-size=0` , it'd be better to let the command fail with an informative error message (as done in the ensuing `if` block) rather than silently ignore the flag and set `mnemonicSize` to 24.

Reply:
I've added it due to the fact that tests would fail without it. I didn't understand why, as it should pick up the default value of 24 words, but it did :confused: 

I will however remove it, and we will look into it together if you agree :smile:

Reply:
I've removed it, and test are now failing. Can you take a look at what's wrong @alessio?